### PR TITLE
use CUDA 13 for third-party integration tests

### PR DESCRIPTION
**Pull request by jameslamb**
## Description

Contributes to https://github.com/rapidsai/build-planning/issues/208

Reverts #19903 and #19917 

Now that there are `xgboost` conda packages with CUDA 13 support (https://github.com/rapidsai/xgboost-feedstock/issues/100), switches the `third-party-integration-tests-cudf-pandas` to use CUDA 13.

This also switches to the more generic CI Image tags (`rapidsai/26.02-latest`), so that configuration doesn't need to be updated as RAPIDS adds support for new minor  versions of CUDA.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by jameslamb** [comment]
I think this is working... the `third-party-integration-tests-cudf-pandas` job is passing: https://github.com/rapidsai/cudf/actions/runs/19827202930/job/56806558906?pr=20748

Other failures look like network or Anaconda infrastructure issues, which would hopefully be resolved with a re-run:

> critical libmamba Multiple errors occurred:
    Download error (28) Timeout was reached [https://conda.anaconda.org/conda-forge/noarch/repodata.json.zst]
    Operation too slow. Less than 30 bytes/sec transferred the last 60 seconds
    Subdir conda-forge/noarch not loaded!
    If you run into this error repeatedly, your package cache may be corrupted.
    Please try running `mamba clean -a` to remove this cache before retrying the operation.
>    If you still are having issues, please report the error on `mamba-org/mamba`'s issue tracker:
    https://github.com/mamba-org/mamba/issues/new?assignees=&labels=&projects=&template=bug.yml
> [rapids-conda-retry] conda returned exit code: 1
> [rapids-conda-retry] Exiting, reached max retries...

([conda-python-other-tests link](https://github.com/rapidsai/cudf/actions/runs/19827202930/job/56806570597?pr=20748#step:11:238))

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -42,11 +42,11 @@ jobs:
         # This gate is here and not at the job level because we need the job to not be skipped,
         # since other jobs depend on it.
         if: ${{ vars.TELEMETRY_ENABLED == 'true' }}
-        uses: rapidsai/shared-actions/telemetry-dispatch-stash-base-env-vars@main
+        uses: rapidsai/shared-actions/telemetry-dispatch-stash-base-env-vars@no-proxy-cache
```
Comment (jameslamb) [inline .github/workflows/build.yaml]: TODO before merge: revert all of these

Testing conda timeouts, with the changes from https://github.com/rapidsai/shared-workflows/pull/472

**Comment by vyasr** [comment]
I've had https://github.com/rapidsai/cudf/pull/19948 open for a while on this but was stalled waiting for Pytorch. In the interim (last week) we merged https://github.com/rapidsai/cudf/pull/20729, which means that we now accept [installing CPU pytorch](https://github.com/rapidsai/cudf/actions/runs/19839750040/job/56847199679?pr=20748#step:12:11988) since we then just [skip those tests](https://github.com/rapidsai/cudf/actions/runs/19839750040/job/56847199679?pr=20748#step:12:12683). @Matt711 you pretty much said [you're fine with this outcome](https://github.com/rapidsai/cudf/pull/20729#issuecomment-3576935883), right? Like @bdice [said](https://github.com/rapidsai/cudf/pull/20729#pullrequestreview-3507291674), there's some risk of us just not noticing happening, but I agree that it's a net benefit overall.

**Comment by jameslamb** [comment]
Ah thanks @vyasr , sorry I missed that already-open PR.

I've removed the other testing bits from this, it should be ready now.

**Review by bdice — APPROVED** [review]
